### PR TITLE
feat(content-conformance): detect partials

### PR DESCRIPTION
**Pull request by BRKalow**
🎟️ [Asana Task]()

---

## Description

Detect if a file is a partial when initializing `ContentFile` and sets `isPartial` appropriately.

## PR Checklist 🚀

- [ ] Conduct thorough self-review.
- [ ] Add or update tests as appropriate.
- [ ] Write a useful description (above) to give reviewers appropriate context.
- [ ] Identify (in the description above) and document (add Asana tasks on [this board](https://app.asana.com/0/1100423001970639/list)) any technical debt that you're aware of, but are not addressing as part of this PR.

### CODE REVIEW

File: packages/content-conformance/src/config.ts [hunk 1]
```diff
@@ -64,6 +65,10 @@ export async function loadConfig({
 
   config.root = resolveConfigRoot(config.root, cwd)
 
+  if (!config.partialsDirectory) {
+    config.partialsDirectory = 'content/partials'
```
Comment (thiskevinwang) [inline packages/content-conformance/src/config.ts:69]: Is this a baked in default?

Comment (BRKalow) [reply packages/content-conformance/src/config.ts:69]: Yeah, it seemed like a reasonable default. Think we should not do it here?

Comment (thiskevinwang) [reply packages/content-conformance/src/config.ts:69]: I think here is OK, but also feel like a central file/export of **all-things-defaults** could be useful for preventing _default config sprawl_ (very loose opinion)

Comment (thiskevinwang) [reply packages/content-conformance/src/config.ts:69]: Additionally, curious if this needs to account for 'legacy' `pages/partials` 
- https://github.com/hashicorp/vault/tree/v1.4.0/website/pages/partials

Comment (BRKalow) [reply packages/content-conformance/src/config.ts:69]: Good callout re legacy partials. I'm thinking probably not, but if it turns out that folks are editing content that far back we might want to handle it.

Comment (BRKalow) [reply packages/content-conformance/src/config.ts]: @thiskevinwang I abstracted out the default values into a constant at the top of `config.ts`, but holding off on a separate file until we start introducing more default values used throughout the package.

**Review by thiskevinwang — APPROVED** [review]
👍

**Pull request merged**